src/main/kotlin/com/hyperdevs/poeditor/gradle/PoEditorStringsImporter.kt

```diff
@@ -80,18 +82,41 @@ object PoEditorStringsImporter {
                               defaultLang: String,
                               resDirPath: String,
                               tags: List<String>?,
-                              languageValuesOverridePathMap: Map<String, String>?) {
+                              languageValuesOverridePathMap: Map<String, String>?,
+                              minimumTranslationPercentage: Int?) {
+        fun isPercentageTooLow(percentage: Double) = percentage < minimumTranslationPercentage ?: -1
+        fun List<ProjectLanguage>.joinAndFormat(transform: ((ProjectLanguage) -> CharSequence)) =
+            joinToString(separator = ", ", prefix = "[", postfix = "]", transform = transform)
+
         try {
             val poEditorApiController = PoEditorApiControllerImpl(apiToken, poEditorApi)
 
             // Retrieve available languages from PoEditor
             logger.lifecycle("Retrieving project languages...")
             val projectLanguages = poEditorApiController.getProjectLanguages(projectId)
+            val skippedLanguages = projectLanguages.filter { language -> isPercentageTooLow(language.percentage) }
 
             // Iterate over every available language
-            logger.lifecycle("Available languages: [${projectLanguages.joinToString(", ") { it.code }}]")
+            logger.lifecycle("Available languages: ${projectLanguages.joinAndFormat { it.code }}")
+
+            minimumTranslationPercentage?.run {
+                val skippedLanguagesMessage = if (skippedLanguages.isEmpty()) {
+                    "All languages are translated above the minimum of $minimumTranslationPercentage%"
+                } else {
+                    val formattedLanguages = skippedLanguages.joinAndFormat { "${it.code} (${it.percentage}%)" }
+                    "Skipping languages translated under $minimumTranslationPercentage%: $formattedLanguages"
+                }
+
+                logger.lifecycle(skippedLanguagesMessage)
+            }
+
             projectLanguages.forEach { languageData ->
                 val languageCode = languageData.code
+                val percentage = languageData.percentage
+
+                if (isPercentageTooLow(percentage)) {
+                    return@forEach
+                }
```

Comment:
If you don't want to rerun the `isPercentageTooLow` function you could do something like:
```kotlin
if (skippedLanguages.any { it.code == languageCode }) {
  return@forEach
}
```
And check against the already calculated languages to skip.

BTW, not sure about this, but would the `return@forEach` exit the `forEach` completely without iterating the rest of the languages, or would it act as a `continue` clause used in standard `for` loops?

Reply:
OK, I'll do it this way!
Initially I preferred to not do this because it makes the assumption that each returned language has a unique language code etc. But it's a safe assumption 👍

return@forEach is like a **continue**, there's no direct equivalent for **break** (but there's a workaround).
It just returns from the lambda expression, so all languages will be checked.

Reply:
Looks good to me! And thanks for the clarifications!